server/routes/helpers.js

```diff
@@ -88,24 +88,25 @@ const getPaginationObject = (pageParams) => {
   }
 
   for (let i = startNum; i <= endNum; i++) {
+    const href = `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${i}`
     pageItems.push({
       text: i,
-      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${i}`,
+      href: showAllMatches ? `${href}&showAllMatches=True` : href,
       selected: currentPage === i
     })
   }
 
   if (currentPage !== 1) {
     previousLink = {
       text: 'Previous',
-      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${currentPage - 1}`
+      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${currentPage - 1}${showAllMatches ? '&showAllMatches=True' : ''}`
     }
   }
 
   if (currentPage < totalPages) {
     nextLink = {
       text: 'Next',
-      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${currentPage + 1}`
+      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${currentPage + 1}${showAllMatches ? '&showAllMatches=True' : ''}`
     }
```

Comment:
Can we just do this if condition once instead of 3 times?

Reply:
I did not get this comment tbh. It is being used where required

Reply:
```
const allMatchesShown=showAllMatches ? '&showAllMatches=True' : ''
 if (currentPage < totalPages) {
    nextLink = {
      text: 'Next',
      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${currentPage + 1}${allMatchesShown}`
    }
```

Reply:
It will be a bit easier to maintain that way. Really we should do the same with the href which is always the same apart from the page number

Reply:
The idea is not to have 3 checks for whether `showAllMatches` is `true`, all of which result in the same decision being made

Reply:
Thanks, have made the changes as per the above suggestion